From 99f2c84a457bac958678aa60ef795f22948aa382 Mon Sep 17 00:00:00 2001
Message-Id: <99f2c84a457bac958678aa60ef795f22948aa382.1426052288.git.chang-joon.lee@intel.com>
From: sudheer gurram <sudheer.gurram@intel.com>
Date: Fri, 13 Feb 2015 09:04:39 +0530
Subject: [PATCH 1/5] SQUASHME! [VPG]: drm/i915: Remove unnecessary mutex
 around stats tracking.

The mutex lock around gem obj insert pid & gem obj remove pid are not needed as
the gem handle will not be freed as those functions are called during handle
creation & destruction.

Squash this patch to:
commit e3b5c17c738b28c741e04a8429adae426f679bc1
Author: Sourab Gupta <sourab.gupta@intel.com>
Date:   Mon Aug 11 13:12:04 2014 +0530

    MUST_REBASE [VPG]: drm/i915: Sysfs interface to get GFX shmem usage
    stats per process

Issue: GMINL-6552
Change-Id: I8d18fde4c41fde6f54226a073c1ddf97bc5ea682
Signed-off-by: sudheer gurram <sudheer.gurram@intel.com>
---
 drivers/gpu/drm/i915/i915_gem_debug.c |   17 +++--------------
 1 file changed, 3 insertions(+), 14 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_debug.c b/drivers/gpu/drm/i915/i915_gem_debug.c
index 80ac0d9..988693f 100644
--- a/drivers/gpu/drm/i915/i915_gem_debug.c
+++ b/drivers/gpu/drm/i915/i915_gem_debug.c
@@ -269,10 +269,6 @@ int i915_gem_obj_insert_pid(struct drm_i915_gem_object *obj)
 	if (!i915.memtrack_debug)
 		return 0;
 
-	ret = i915_mutex_lock_interruptible(obj->base.dev);
-	if (ret)
-		return ret;
-
 	list_for_each_entry(entry, &obj->pid_info, head) {
 		if (entry->tgid == current_tgid) {
 			entry->open_handle_count++;
@@ -284,16 +280,14 @@ int i915_gem_obj_insert_pid(struct drm_i915_gem_object *obj)
 		entry = kzalloc(sizeof(*entry), GFP_KERNEL);
 		if (entry == NULL) {
 			DRM_ERROR("alloc failed\n");
-			ret = -ENOMEM;
-			goto out;
+			return -ENOMEM;
 		}
 		entry->tgid = current_tgid;
 		entry->open_handle_count = 1;
 		INIT_LIST_HEAD(&entry->virt_addr_head);
 		list_add_tail(&entry->head, &obj->pid_info);
 	}
-out:
-	mutex_unlock(&obj->base.dev->struct_mutex);
+
 	return ret;
 }
 
@@ -302,15 +296,11 @@ void i915_gem_obj_remove_pid(struct drm_i915_gem_object *obj)
 	pid_t current_tgid = task_tgid_nr(current);
 	struct drm_i915_obj_pid_info *pid_entry, *pid_next;
 	struct drm_i915_obj_virt_addr *virt_entry, *virt_next;
-	int ret, found = 0;
+	int found = 0;
 
 	if (!i915.memtrack_debug)
 		return;
 
-	ret = i915_mutex_lock_interruptible(obj->base.dev);
-	if (ret)
-		return;
-
 	list_for_each_entry_safe(pid_entry, pid_next, &obj->pid_info, head) {
 		if (pid_entry->tgid == current_tgid) {
 			pid_entry->open_handle_count--;
@@ -329,7 +319,6 @@ void i915_gem_obj_remove_pid(struct drm_i915_gem_object *obj)
 			break;
 		}
 	}
-	mutex_unlock(&obj->base.dev->struct_mutex);
 
 	if (found == 0)
 		DRM_DEBUG("Couldn't find matching tgid %d for obj %p\n",
-- 
1.7.9.5

